src/k8s/pkg/k8sd/api/kubeconfig.go

```diff
@@ -1,25 +1,39 @@
 package api
 
 import (
+	"encoding/json"
 	"fmt"
 	"net/http"
-	"os"
 
 	apiv1 "github.com/canonical/k8s/api/v1"
 	"github.com/canonical/lxd/lxd/response"
 	"github.com/canonical/microcluster/state"
+	"k8s.io/client-go/tools/clientcmd"
 )
 
 func getKubeconfig(s *state.State, r *http.Request) response.Response {
-	// TODO: Render a new kubeconfig instead of reading the existing one
-	//       when the config can be altered via request parameters.
-	config, err := os.ReadFile("/etc/kubernetes/admin.conf")
+	req := apiv1.GetKubeConfigRequest{}
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		return response.BadRequest(fmt.Errorf("failed to parse request: %w", err))
+	}
+	server := req.Server
+	if req.Server == "" {
+		server = fmt.Sprintf("https://%s:6443", s.Address().Hostname())
+	}
+
+	config := clientcmd.GetConfigFromFileOrDie("/etc/kubernetes/admin.conf")
```

Comment:
NOTE: `GetConfigFromFileOrDie` will exit in case the file cannot be read. This is not what we want here, we want to handle possible errors instead.